src/view/gui/fMainWindow.cpp

```diff
@@ -240,41 +240,41 @@ fMainWindow::fMainWindow()
-  m_tabWidget->setMinimumHeight(minheight);
-  m_tabWidget->setMaximumHeight(m_tabWidget->minimumHeight());
-
-  m_toolTabdock->setFeatures(QDockWidget::DockWidgetFloatable);
-  m_toolTabdock->setWidget(m_tabWidget);
-  this->addDockWidget(Qt::TopDockWidgetArea, m_toolTabdock);
-  this->m_toolTabdock->setWindowTitle("Double click to undock");
+  //m_toolTabdock = new CaPTkDockWidget(this); // custom class to propagate drag-and-drop events to the main window
+  //m_toolTabdock->setWindowFlags(Qt::SubWindow); // SubWindow allows it to be shown while MainWindow is also visible
+
+  //m_tabWidget = new QTabWidget(m_toolTabdock);
+  //infoPanel = new fBottomImageInfoTip(centralwidget);
+  //imagesPanel = new fImagesPanel(m_tabWidget); // New Images Panel
+  //m_tabWidget->addTab(imagesPanel, QString());
+  //tumorPanel = new fTumorPanel(m_tabWidget);
+  //m_tabWidget->addTab(tumorPanel, QString());
+  //drawingPanel = new fDrawingPanel(m_tabWidget);
+  //featurePanel = new fFeaturePanel(m_tabWidget);
+  //m_tabWidget->addTab(drawingPanel, QString());
+  //m_tabWidget->addTab(featurePanel, "Feature Extraction");
+  //int minheight = /*std::max(drawingPanel->sizeHint().height(), featurePanel->sizeHint().height())*/featurePanel->sizeHint().height() + 25;
+  //m_tabWidget->setMinimumHeight(minheight);
+  //m_tabWidget->setMaximumHeight(m_tabWidget->minimumHeight());
+
+  //m_toolTabdock->setFeatures(QDockWidget::DockWidgetFloatable);
+  //m_toolTabdock->setWidget(m_tabWidget);
+  //this->addDockWidget(Qt::TopDockWidgetArea, m_toolTabdock);
+  //this->m_toolTabdock->setWindowTitle("Double click to undock");
 
   // Set up our connections so that fMainWindow can receive all drag-and-drop events from our tool tab dock
-  connect(m_toolTabdock, SIGNAL(dragEnteredDockWidget(QDragEnterEvent*)), this, SLOT(dragEnterEvent(QDragEnterEvent*)));
-  connect(m_toolTabdock, SIGNAL(droppedOnDockWidget(QDropEvent*)), this, SLOT(dropEvent(QDropEvent*)));
+  //connect(m_toolTabdock, SIGNAL(dragEnteredDockWidget(QDragEnterEvent*)), this, SLOT(dragEnterEvent(QDragEnterEvent*)));
+  //connect(m_toolTabdock, SIGNAL(droppedOnDockWidget(QDropEvent*)), this, SLOT(dropEvent(QDropEvent*)));
 
   //! automatic undock on low resolution
-  //! to be tested thoroughly
-  QScreen *scr = QGuiApplication::primaryScreen();
-  //!if primary screen resolution is lower than 1200x1024(any of x,y values)
-  if ((scr->size().width() < 1200) || (scr->size().height() < 1024))
-  {
-	  this->m_toolTabdock->setWindowTitle("Double click to dock");
-	  this->m_toolTabdock->setFloating(true);
-  }
+  ////! to be tested thoroughly
+  //QScreen *scr = QGuiApplication::primaryScreen();
+  ////!if primary screen resolution is lower than 1200x1024(any of x,y values)
+  //if ((scr->size().width() < 1200) || (scr->size().height() < 1024))
+  //{
+	 // this->m_toolTabdock->setWindowTitle("Double click to dock");
+	 // this->m_toolTabdock->setFloating(true);
+  //}
```

Comment:
Perhaps you forgot to uncomment this block. This is needed for tabwidget functionality.

Reply:
uncommented this block and committed the code.